play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/EvaluationSettings.kt

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.wear.watchface.dfx.memory
+
+import com.google.common.collect.ImmutableList
+import java.util.Optional
+import java.util.jar.Manifest
+import kotlin.system.exitProcess
+import org.apache.commons.cli.DefaultParser
+import org.apache.commons.cli.HelpFormatter
+import org.apache.commons.cli.Option
+import org.apache.commons.cli.Options
+import org.apache.commons.cli.ParseException
+
+/**
+ * Contains the CLI arguments that the script was invoked with.
+ *
+ * @property watchFacePath Path to the watch face package to be evaluated.
+ * @property schemaVersion Watch Face Format schema version that the evaluation should use.
+ */
+class EvaluationSettings(
+    val watchFacePath: String,
+    val schemaVersion: String,
+) {
+    constructor(
+        watchFacePath: String,
+        schemaVersion: String,
+        greedyEvaluationSwitch: Int,
+    ) : this(watchFacePath, schemaVersion) {
+        this.greedyEvaluationSwitch = greedyEvaluationSwitch
+    }
+    /**
+     * The maximum number of configurations of a watch face under which the real footprint is
+     * calculated. If a watch face has more than this number of configurations, the greedy
+     * evaluations is used.
+     */
+    var greedyEvaluationSwitch: Int = GREEDY_DEFAULT_LIMIT
+        private set
+
+    /** The memory limit in bytes for the ambient mode. Defaults to 10 * 10^20 bytes (or 10 MB). */
+    var ambientLimitBytes: Long = MemoryFootprint.toBytes(10.0)
+        private set
+    /** The memory limit in bytes for the active mode. Defaults to 10 * 10^20 bytes (or 10 MB). */
+    var activeLimitBytes: Long = MemoryFootprint.toBytes(10.0)
```

Comment:
A developer made the comment recently, is it sensible to change this default to 100, to align with the Play default? Is it worth doing this as part of the migration?